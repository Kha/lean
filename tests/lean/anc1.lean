#check (⟨1, 2⟩ : nat × nat)

#check (⟨trivial, trivial⟩ : true ∧ true)

example : true := sorry

#check (⟨1, sorry⟩ : Σ' x : nat, x > 0)

open tactic

#check show true, from ⟨⟩

#check (⟨1, by intro1 >> contradiction⟩ : ∃ x : nat, 1 ≠ 0)
universe variables u v
#check λ (A B C : Prop),
  assume (Ha : A) (Hb : B) (Hc : C),
  show B ∧ A, from
  ⟨Hb, Ha⟩

#check λ (A B C : Prop),
  assume (Ha : A) (Hb : B) (Hc : C),
  show B ∧ A ∧ C ∧ A, from
  ⟨Hb, ⟨Ha, ⟨Hc, Ha⟩⟩⟩

#check λ (A B C : Prop),
  assume (Ha : A) (Hb : B) (Hc : C),
  show B ∧ A ∧ C ∧ A, from
  ⟨Hb, Ha, Hc, Ha⟩

#check λ (A B C : Prop),
  assume (Ha : A) (Hb : B) (Hc : C),
  show ((B ∧ true) ∧ A) ∧ (C ∧ A), from
  ⟨⟨⟨Hb, ⟨⟩⟩, Ha⟩, ⟨Hc, Ha⟩⟩

#check λ (A : Type u) (P : A → Prop) (Q : A → Prop),
  assume (a : A), assume (H1 : P a) (H2 : Q a),
  show ∃ x, P x ∧ Q x, from
  ⟨a, ⟨H1, H2⟩⟩

#check λ (A : Type u) (P : A → Prop) (Q : A → Prop),
  assume (a : A) (b : A), assume (H1 : P a) (H2 : Q b),
  show ∃ x y, P x ∧ Q y, from
  ⟨a, ⟨b, ⟨H1, H2⟩⟩⟩

#check λ (A : Type u) (P : A → Prop) (Q : A → Prop),
  assume (a : A) (b : A), assume (H1 : P a) (H2 : Q b),
  show ∃ x y, P x ∧ Q y, from
  ⟨a, b, H1, H2⟩
