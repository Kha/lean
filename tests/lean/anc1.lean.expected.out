(1, 2) : ℕ × ℕ
and.intro trivial trivial : true ∧ true
anc1.lean:5:0: warning: declaration '[anonymous]' uses sorry
⟨1, sorry⟩ : Σ' (x : ℕ), x > 0
show true, from true.intro : true
Exists.intro 1 (id_locked (1 ≠ 0) (λ (a : 1 = 0), nat.no_confusion a)) : ∃ (x : ℕ), 1 ≠ 0
λ (A B C : Prop) (Ha : A) (Hb : B) (Hc : C), show B ∧ A, from and.intro Hb Ha :
  ∀ (A B C : Prop), A → B → C → B ∧ A
λ (A B C : Prop) (Ha : A) (Hb : B) (Hc : C),
  show B ∧ A ∧ C ∧ A, from and.intro Hb (and.intro Ha (and.intro Hc Ha)) :
  ∀ (A B C : Prop), A → B → C → B ∧ A ∧ C ∧ A
λ (A B C : Prop) (Ha : A) (Hb : B) (Hc : C),
  show B ∧ A ∧ C ∧ A, from and.intro Hb (and.intro Ha (and.intro Hc Ha)) :
  ∀ (A B C : Prop), A → B → C → B ∧ A ∧ C ∧ A
λ (A B C : Prop) (Ha : A) (Hb : B) (Hc : C),
  show ((B ∧ true) ∧ A) ∧ C ∧ A, from and.intro (and.intro (and.intro Hb true.intro) Ha) (and.intro Hc Ha) :
  ∀ (A B C : Prop), A → B → C → ((B ∧ true) ∧ A) ∧ C ∧ A
λ (A : Type u) (P Q : A → Prop) (a : A) (H1 : P a) (H2 : Q a),
  show ∃ (x : A), P x ∧ Q x, from Exists.intro a (and.intro H1 H2) :
  ∀ (A : Type u) (P Q : A → Prop) (a : A), P a → Q a → (∃ (x : A), P x ∧ Q x)
λ (A : Type u) (P Q : A → Prop) (a b : A) (H1 : P a) (H2 : Q b),
  show ∃ (x y : A), P x ∧ Q y, from Exists.intro a (Exists.intro b (and.intro H1 H2)) :
  ∀ (A : Type u) (P Q : A → Prop) (a b : A), P a → Q b → (∃ (x y : A), P x ∧ Q y)
λ (A : Type u) (P Q : A → Prop) (a b : A) (H1 : P a) (H2 : Q b),
  show ∃ (x y : A), P x ∧ Q y, from Exists.intro a (Exists.intro b (and.intro H1 H2)) :
  ∀ (A : Type u) (P Q : A → Prop) (a b : A), P a → Q b → (∃ (x y : A), P x ∧ Q y)
